File: backend/src/main/java/com/mini/credit/dto/credit/AnalyseRisqueRequest.java
package com.mini.credit.dto.credit;

import com.mini.credit.enums.NiveauRisque;
import com.mini.credit.enums.RecommandationRisque;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Setter
public class AnalyseRisqueRequest {

    private String analyseId;

    private LocalDate dateVisite;
    private String lieuVisite;
    private Boolean activiteVerifiee;
    private String descriptionActivite;
    private String ancienneteActivite;
    private BigDecimal chiffreAffairesEstime;
    private String chiffreAffairesDevise;
    private BigDecimal revenuNetEstime;
    private String revenuNetDevise;
    private BigDecimal chargesMensuelles;
    private String chargesMensuellesDevise;
    private BigDecimal capaciteRemboursement;
    private String capaciteRemboursementDevise;
    private String montantDemandeDevise;
    private String fraisDemandeDevise;
    private String depotRequisDevise;
    private String depotPayeDevise;
    private NiveauRisque risqueNiveau;
    private BigDecimal scoreRisque;
    private Boolean scoreRisqueCorrigeManuellement;

    @NotNull
    private RecommandationRisque recommandation;

    private String commentaire;
}

File: frontend/src/app/features/credit/models/analyse-risque-request.ts
import { NiveauRisque } from "./enums/niveau-risque.enum";
import { RecommandationRisque } from "./enums/recommandation-risque.enum";

export interface AnalyseRisqueRequest {
  analysteId?: number;
  dateVisite?: string;
  lieuVisite?: string;
  activiteVerifiee?: boolean;
  descriptionActivite?: string;
  ancienneteActivite?: string;
  chiffreAffairesEstime?: number;
  chiffreAffairesDevise?: string;
  revenuNetEstime?: number;
  revenuNetDevise?: string;
  chargesMensuelles?: number;
  chargesMensuellesDevise?: string;
  capaciteRemboursement?: number;
  capaciteRemboursementDevise?: string;
  montantDemandeDevise?: string;
  fraisDemandeDevise?: string;
  depotRequisDevise?: string;
  depotPayeDevise?: string;
  risqueNiveau?: NiveauRisque;
  scoreRisque?: number;
  scoreRisqueCorrigeManuellement?: boolean;
  recommandation: RecommandationRisque;
  commentaire?: string;
}